easybuild/easyblocks/generic/pythonpackage.py

```diff
@@ -655,10 +656,26 @@ def run(self, *args, **kwargs):
         super(PythonPackage, self).run(*args, **kwargs)
 
         # configure, build, test, install
-        self.configure_step()
-        self.build_step()
-        self.test_step()
-        self.install_step()
+        # See EasyBlock.get_steps
+        steps = [
+            (CONFIGURE_STEP, 'configuring', [lambda x: x.configure_step], True),
+            (BUILD_STEP, 'building', [lambda x: x.build_step], True),
+            (TEST_STEP, 'testing', [lambda x: x.test_step], True),
+            (INSTALL_STEP, "installing", [lambda x: x.install_step], True),
+        ]
+        self.skip = False  # --skip does not apply here
+        self.silent = build_option('silent')
+        # See EasyBlock.run_all_steps
+        for (step_name, descr, step_methods, skippable) in steps:
+            if self._skip_step(step_name, skippable):
```

Comment:
Methods that start with `_` are considered not to be a part of the public API (by convention, in Python code).

To avoid trouble, like changing or renaming `_skip_step` which may break the usage, we should provide a proper `skip_step` method in framework (which can be a trivial wrapper around `_skip_step`, or a proper rename of `_skip_step` with some minimal cleanup where needed).

Reply:
Rename of `_skip_step` to `skip_step` done in https://github.com/easybuilders/easybuild-framework/pull/3561

Reply:
I was aware of that, but in this case this is an easybuild package too, so we could just do that.
Also this method being private may also just mean that it is not to be used outside the class. In this case we are inside the class (a descendant) which is IIRC fully allowed, but well... The PR is open anyway, so I'll change this here